pageserver/src/walreceiver/connection_manager.rs

```diff
@@ -547,7 +620,7 @@ impl WalreceiverState {
     /// Optionally, omits the given node, to support gracefully switching from a healthy safekeeper to another.
     ///
     /// The candidate that is chosen:
-    /// * has fewest connection attempts from pageserver to safekeeper node (reset every time we receive a WAL message from the node)
+    /// * has no pending retry cooldown
```

Comment:
`next_connection_candidate`'s rustdoc now needs some adjustments too, at least

> pick a candidate ... and lowest failed connection attemps

needs removal/rephrasing.

Reply:
Fixed, but I found writing comments about new retry logic somehow hard and may have missed some places. If you know other places with inappropriate or missing comments, let me know.